src/pip/_internal/wheel_builder.py

```diff
@@ -116,7 +115,7 @@ def _should_cache(
     has determined a wheel needs to be built.
     """
     if not should_build_for_install_command(
-        req, check_binary_allowed=check_binary_allowed
+        req, check_binary_allowed=_always_true
     ):
```

Comment:
Side note: this test actually boils down to not caching wheels built from editable requirements.